data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -185,16 +204,29 @@ private CloseableIterable<T> applyEqDeletes(CloseableIterable<T> records) {
         .reduce(Predicate::and)
         .orElse(t -> true);
 
-    Filter<T> remainingRowsFilter = new Filter<T>() {
+    if (hasColumnIsDeleted) {
+      return CloseableIterable.transform(records, record -> {
```

Comment:
I think these two can also be abstracted to `Deletes`
```java
  public static <T> CloseableIterable<T> filter(CloseableIterable<T> rows, Predicate<T> pred, Consumer<T> consumer) {
    return CloseableIterable.transform(rows, row -> {
      if (!pred.test(row)) {
        consumer.accept(row);
      }
      return row;
    });
  }
```

Reply:
Nice suggestion. Made the change.